### PR TITLE
[sled-agent] Update propolis dependency, to use simpler paths

### PR DESCRIPTION
**Pull request by smklein**
Depends on https://github.com/oxidecomputer/propolis/pull/117

### CODE REVIEW

File: package-manifest.toml [hunk 1]
```diff
@@ -89,5 +89,5 @@ zone = true
 [external_package.propolis-server.source]
 type = "prebuilt"
 repo = "propolis"
-commit = "e984409a425df97338efa2c53ba3e40c73668638"
-sha256 = "206ef42e90d157e3314f4745ddb4422b3c93a16881749824037853e6f39f3762"
+commit = "7f370d295813be75160b5e6ecc79f6d3c7ed9368"
+sha256 = "32994c25bc7481f9aeb3e0976425cf39a6ee6fd8a501929371884d9add53d310"
```
Comment (smklein) [inline package-manifest.toml:93]: heh, we're publishing artifacts aggressively enough that these exist, even for the branch that hasn't been merged yet.

TODO(me-before-I-merge-this): Update these values - and the rev in the `Cargo.toml` - to the merged versions

File: sled-agent/src/instance.rs [hunk 2]
```diff
@@ -92,7 +92,7 @@ async fn wait_for_http_server(
             // This request is nonsensical - we don't expect an instance to be
             // using the nil UUID - but getting a response that isn't a
```
Comment (luqmana) [inline sled-agent/src/instance.rs:93]: Update comment

**Review by luqmana — APPROVED** [review]
💯 Thanks Sean!

File: sled-agent/src/instance.rs (hunk 2 above)
Comment (smklein) [reply sled-agent/src/instance.rs]: Done

**Pull request merged**